app/models/services/crossref.rb

```diff
@@ -0,0 +1,24 @@
+class Crossref
+  def initialize(options={})
+    @base_url = 'http://api.crossref.org/'
+    @metadata_url = @base_url + "works/"
+    @funder_url = @base_url + "funders/"
+  end
+
+  def get_doi_metadata(doi)
+    metadata_uri = URI.parse(@metadata_url+doi)
+    JSON.parse Net::HTTP.get(metadata_uri) if valid_json? Net::HTTP.get(metadata_uri)
+  end
+
+  private
+
+    def valid_json?(json)
+      begin
+        JSON.parse(json)
+        return true
+      rescue JSON::ParserError => e
+        return false
+      end
+    end
```

Comment:
You mean run the rescue there?  I got valid from a stackoverflow response so I only kinda know what it does, fault of my ignorance.

Something like:

```
JSON.parse Net::HTTP.get(metadata_uri)
rescue JSON::ParserError => e
```